### PR TITLE
helper tool to track XenServer components within nixpkgs

### PR DESCRIPTION
**Pull request by ts468**

### CODE REVIEW

File: pkgs/applications/virtualization/xenserver-buildroot/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{stdenv, fetchurl, fetchgit, nix-template-rpm }:
+
+# Xenserver buildroot of somehow around XenServer 6.5.
```
Comment (falsifian) [inline pkgs/applications/virtualization/xenserver-buildroot/default.nix]: This comment might make a good meta.longDescription.

File: pkgs/applications/virtualization/xenserver-buildroot/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    #
+    # generate script to update buildRootInput.nix
+    #
+    mkdir -p $out/bin
+    cat << EOF > $out/bin/xapi-generate-nix-buildRootInput
+    #!$SHELL
+    if [ -e nixpkgs ] || [ -e buildRootInput.nix ] || [ -e translation.table ]
+    then
+      echo "Error: directory 'nixpkgs', file 'buildRootInput.nix' or file 'translation.table' does already exist!"
```
Comment (falsifian) [inline pkgs/applications/virtualization/xenserver-buildroot/default.nix]: "already exists" is more common phrasing than "does already exist". (The latter is easy to confuse for "doesn't exist", at least for me.)

File: pkgs/applications/virtualization/xenserver-buildroot/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    url = "https://github.com/xenserver/buildroot";
+    rev = "5e9892582c3733416bf8b6ebf9aece8cefa15b28";
+    sha256 = "6ee4288fb715e1315ff0cc7923ec1efe896be176d6697a7d56015b77b4317447";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  buildInputs = [ nix-template-rpm ];
+
+    #cp ${ ./buildRootInput.nix } $out/usr/share/buildroot/buildRootInput.nix
```
Comment (falsifian) [inline pkgs/applications/virtualization/xenserver-buildroot/default.nix]: Did you mean to leave this here?

File: pkgs/applications/virtualization/xenserver-buildroot/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  installPhase = ''
+    mkdir -p $out/usr/share/buildroot
+    cp -r SPECS $out/usr/share/buildroot/
+    cp -r SOURCES $out/usr/share/buildroot/
+    cp ${ ./translation.table } $out/usr/share/buildroot/translation.table
+
+    #
+    # generate script to update buildRootInput.nix
```
Comment (falsifian) [inline pkgs/applications/virtualization/xenserver-buildroot/default.nix]: Could you add documentation somewhere that explains what buildRootInput.nix is? Three questions come to mind:
* Is it a file somewhere in the nixpkgs repository?
* What does it build?
* What does it mean to update it? (Does it fetch the latest version from some website?)

**Comment by falsifian** [comment]
I added some minor comments.

I don't have the proper context to review this properly, but I guess it can't hurt to merge it in since it doesn't change any existing stuff.

Can you at least confirm you've tested it and it works as expected?

**Comment by ts468** [comment]
@falsifian, thank you very much for the detailed review! I'll address all the points that you raised.

**Comment by falsifian** [comment]
Thanks. The most important thing: can you confirm you've used this expression and it works?

The other points are minor and can be fixed after merging this.

File: pkgs/applications/virtualization/xenserver-buildroot/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{stdenv, fetchurl, fetchgit, nix-template-rpm }:
+
+# Xenserver buildroot of somehow around XenServer 6.5.
+# Xenserver is not an application by itself, but just a collection of components.
+# This package only provides the build inputs to actually build the components of xenserver.
+stdenv.mkDerivation rec {
+  name = "xenserver-${version}";
```
Comment (falsifian) [inline pkgs/applications/virtualization/xenserver-buildroot/default.nix]: Should the name be xenserver-buildroot?

**Comment by ts468** [comment]
@falsifian, thank you very much for your efforts in getting the PR merged. The script works, but part of it depends heavily on the naming of the ocaml modules in PR #8705 and #8672, which is about to change. So once the other two PRs are fixed I would update this PR accordingly before getting it merged.

File: pkgs/applications/virtualization/xenserver-buildroot/default.nix (hunk 5 above)
Comment (ts468) [inline pkgs/applications/virtualization/xenserver-buildroot/default.nix]: Thanks, that would be better, yes.

**Comment by jagajaga** [comment]
ping :)

**Pull request closed without merging**